File: Swift/1-bit-and-2-bit-characters.swift
class Solution {
    func isOneBitCharacter(_ bits: [Int]) -> Bool {
        var parity = 0

        for i in (0..<bits.count-1).reversed() {
            if bits[i] == 0 {
                break
            }
            parity ^= bits[i]
        }

        return parity == 0
    }
}

File: Python/1-bit-and-2-bit-characters.py
class Solution:
    def isOneBitCharacter(self, bits):
        """
        :type bits: List[int]
        :rtype: bool
        """
        parity = 0

        for i in reversed(range(len(bits)-1)):
            if bits[i] == 0:
                break
            parity ^= bits[i]

        return parity == 0
